acceptance/tests/allow_arbitrary_node_name_fact_for_agent.rb

```diff
@@ -11,9 +11,109 @@
 
 node_names.uniq!
 
-authfile = "#{testdir}/auth.conf"
-authconf = node_names.map do |node_name|
-  %Q[
+if @options[:is_puppetserver]
+  step "Ensure legacy auth.conf is disabled for test" do
+    modify_tk_config(master, options['puppetserver-config'], {'jruby-puppet' => {'use-legacy-auth-conf' => false}})
+  end
+
+  teardown do
+    modify_tk_config(master, options['puppetserver-config'], {'jruby-puppet' => {'use-legacy-auth-conf' => true}})
```

Comment:
Seems like this step should just set the value back to whatever it was to start with, which might have been `false`, the new default.  Another option could be to just back up the `puppetserver-config` file in its entirety and restore it as part of the teardown.  I did something like that with the [collect_default_metrics](https://github.com/puppetlabs/puppetserver/blob/97be028c37709701a47abf10ead39d2a152bd32f/acceptance/suites/tests/metrics/collect_default_metrics.rb#L13-L19) test in Puppet Server.